### PR TITLE
Feature/kyv 350 payment changes

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: common/src/main/java/fi/hel/verkkokauppa/common/events/KafkaProducerConfig.java [hunk 1]
```diff
@@ -17,16 +17,16 @@
 public class KafkaProducerConfig {
 
     // add to your application.properties spring.kafka.bootstrap-servers=localhost:9092
-    @Value("${spring.kafka.bootstrap-servers}")
+    @Value("${spring.kafka.bootstrap-servers:#{null}}")
     private String bootstrapServers;
 
-    @Value("${kafka.client.authentication.enabled}")
+    @Value("${kafka.client.authentication.enabled:#{false}}")
```
Comment (hleisma) [inline common/src/main/java/fi/hel/verkkokauppa/common/events/KafkaProducerConfig.java:23]: sane defaults should favour security, need to switch to true

File: common/src/main/java/fi/hel/verkkokauppa/common/rest/CommonServiceConfigurationClient.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private Logger log = LoggerFactory.getLogger(CommonServiceConfigurationClient.class);
+
+    @Autowired
+    private Environment env;
+
+    @Autowired
+    private RestServiceClient restServiceClient;
+
+    public String getRestrictedServiceConfigurationValue(String namespace, String key) {
+        String serviceMappingUrl = env.getRequiredProperty("serviceconfiguration.url") + "restricted/get?namespace=" + namespace + "&key=" + key;
```
Comment (hleisma) [inline common/src/main/java/fi/hel/verkkokauppa/common/rest/CommonServiceConfigurationClient.java:25]: should inject serviceconfiguration.url as class variable (@Value) so it gets read at component startup

File: eventsapi/src/main/java/fi/hel/verkkokauppa/events/configuration/KafkaConsumerConfig.java [hunk 3]
```diff
@@ -23,16 +23,16 @@
 @Configuration
 public class KafkaConsumerConfig {
 
-    @Value("${spring.kafka.bootstrap-servers}")
+    @Value("${spring.kafka.bootstrap-servers:#{null}}")
     private String bootstrapServers;
 
-    @Value("${kafka.client.authentication.enabled}")
+    @Value("${kafka.client.authentication.enabled:#{false}}")
```
Comment (hleisma) [inline eventsapi/src/main/java/fi/hel/verkkokauppa/events/configuration/KafkaConsumerConfig.java:29]: switch to true here too

**Review by hleisma — APPROVED** [review]

File: common/src/main/java/fi/hel/verkkokauppa/common/rest/CommonServiceConfigurationClient.java (hunk 2 above)
Comment (severikupari1) [reply common/src/main/java/fi/hel/verkkokauppa/common/rest/CommonServiceConfigurationClient.java:25]: - [ ] good point, I will add that as @value.

**Pull request merged**